### PR TITLE
marshmallow -> jsonschema

### PR DESCRIPTION
**Pull request by liambuchanan**
* Makes validation reqmts much clearer
* Segregates schema/validation data and code
* Creates a place for commentary on individual fields
* There should be a good way to generate pretty docs though I haven't found anything totally satisfying yet. [matic](https://github.com/mattyod/matic) came closest but it doesn't display comments anywhere in the ui

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          "type": "array",
+          "comment": "An array of namespaces for which to generate jobs. Each namespace here will result in the generation of a job that is identical in all aspects but namespace",
+          "items": {
+            "type": "string"
+          }
+        },
+        "namespaceOverrides": {
+          "type": "object",
+          "comment": "An object mapping namespaces to a set of properties given a generated job belongs to the specified namespace.",
+          "patternProperties": {
```
Comment (imsky) [inline schema.json]: why was patternProperties necessary if the pattern is empty?

Comment (liambuchanan) [reply schema.json]: I don't think there's another way to specify the types of an objects values when the key can be any string

Comment (imsky) [reply schema.json]: would be nice to make sure

**Comment by imsky** [comment]
👍

File: schema.json (hunk 1 above)
Comment (liambuchanan) [reply schema.json]: looks like `additionalProperties` is the way to go

**Comment by liambuchanan** [comment]
thanks @imsky

**Pull request merged**